library/azure_rm_devtestlabslab.py

```diff
@@ -0,0 +1,281 @@
+
+
+class AzureRMDevTestLab(AzureRMModuleBase):
+    """Configuration class for an Azure RM DevTest Lab resource"""
+
+    def __init__(self):
+        self.module_arg_spec = dict(
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str'
+            ),
+            lab_storage_type=dict(
+                type='str',
+                choices=['standard',
+                         'premium'],
+                default='premium'
+            ),
+            premium_data_disks=dict(
+                type='bool'
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.lab = {}
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        super(AzureRMDevTestLab, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                                 supports_check_mode=True,
+                                                 supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        for key in list(self.module_arg_spec.keys()) + ['tags']:
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                self.lab[key] = kwargs[key]
+
+        self.lab['lab_storage_type'] = _snake_to_camel(self.lab['lab_storage_type'], True)
+        if self.lab.get('premium_data_disks') is not None:
+            self.lab['premium_data_disks'] = 'Enabled' if self.lab['premium_data_disks'] else 'Disabled'
+
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(DevTestLabsClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        resource_group = self.get_resource_group(self.resource_group)
+
+        old_response = self.get_devtestlab()
+
+        if not old_response:
+            self.log("DevTest Lab instance doesn't exist")
+            if self.state == 'absent':
+                self.log("Old instance didn't exist")
+            else:
+                self.to_do = Actions.Create
+        else:
+            self.log("DevTest Lab instance already exists")
+            if self.state == 'absent':
+                self.to_do = Actions.Delete
+            elif self.state == 'present':
+                if self.lab.get('lab_storage_type') != old_response.get('lab_storage_type'):
+                    self.to_do = Actions.Update
+                if self.lab.get('premium_data_disks') != old_response.get('premium_data_disks'):
```

Comment:
is this case sensitive?

Reply:
theoretically no, but fixed it just in case